tests/unit/task-property-test.js

```diff
@@ -0,0 +1,194 @@
+import Ember from 'ember';
+import { task, interval } from 'ember-concurrency';
+import { TaskProperty } from 'ember-concurrency/-task-property';
+import { module, test } from 'qunit';
+
+module('Unit: task property', {
+  beforeEach() {
+    let taskRunCounter = 0;
+    this.waiter = () => taskRunCounter === 0;
+
+    Ember.Test.registerWaiter(this.waiter);
+
+    TaskProperty.prototype.withTestWaiter = function() {
+      if (Ember.testing) {
+        let originalTaskFn = this.taskFn;
+
+        this.taskFn = function * (...args) {
+          taskRunCounter += 1;
+          try {
+            return yield * originalTaskFn.apply(this, args);
+          } finally {
+            taskRunCounter -= 1;
+          }
+        };
+      }
+
+      return this;
+    }
+  },
+
+  afterEach() {
+    Ember.Test.unregisterWaiter(this.waiter);
+    delete TaskProperty.prototype.withTestWaiter;
+  }
+});
```

Comment:
Sure, no problem, but maybe only 20 lines of code in this test are specific to the test waiter functionality -- the majority of it is testing that things work as expected in various concurrency scenarios. I originally wrote that code because I wasn't confident in the timing of everything and wanted to prove that things worked as expected -- do you want me to axe all of that too, and basically just keep a modified form of the "withTestWaiter() decorator works" test? Or keep all of them (except the first null hypothesis one which doesn't really make sense with counters), but port them over to counter verification rather than test waiter verification?